.github/scripts/check_diff.py

```diff
@@ -0,0 +1,27 @@
+import json
+import sys
+
+ALL_DIRS = {
+    "libs/core",
+    "libs/community",
+    "libs/langchain",
+    "libs/experimental",
+    "libs/partners/openai",
+}
+
+if __name__ == "__main__":
+    files = sys.argv[1:]
+    dirs_to_run = set()
+
+    for file in files:
+        if any(dir_ in file for dir_ in (".github/workflows", ".github/tools", ".github/actions", "libs/core")):
```

Comment:
might be better to make this a negative case just in case we introduce some new folder structure in the future, probably want to run too much rather than too little CI